store/store.go

```diff
@@ -160,18 +160,36 @@ type Store struct {
 
 var ErrTooManyRequests = errors.New("too many requests")
 
-func respToError(resp *http.Response, msg string) error {
-	if resp.StatusCode == 429 {
-		return ErrTooManyRequests
-	}
+// UnexpectedHTTPStatusError represents an error where the store
+// returned an unexpected HTTP status code.
```

Comment:
What makes an error unexpected? Is it X-Oops-Id header? Can you expand the comment?

Reply:
I'll try, it's the status code that is unexpected,  400 instead of 200, or 404 for something that shouldn't 404 etc.